dowhy/causal_estimators/propensity_score_weighting_estimator.py

```diff
@@ -49,15 +28,14 @@ def __init__(self, *args, min_ps_score=0.05, max_ps_score=0.95, **kwargs):
         self.min_ps_score = min_ps_score
         self.max_ps_score = max_ps_score
 
-    def _estimate_effect(self):
-        psmodel = linear_model.LogisticRegression()
-        psmodel.fit(self._observed_common_causes, self._treatment)
-        self._data['ps'] = psmodel.predict_proba(self._observed_common_causes)[:,1]
-        # trim propensity score weights
-        self._data['ps'] = np.minimum(self.max_ps_score, self._data['ps'])
-        self._data['ps'] = np.maximum(self.min_ps_score, self._data['ps'])
-
-
+    def _estimate_effect(self, recalculate_propensity_score=False):
+        if self._propensity_score_model is None or recalculate_propensity_score is True:
+            self._propensity_score_model = linear_model.LogisticRegression()
+            self._propensity_score_model.fit(self._observed_common_causes, self._treatment)
+            self._data['ps'] = self._propensity_score_model.predict_proba(self._observed_common_causes)[:,1]
+            # trim propensity score weights
```

Comment:
The code for trimming propensity score weights should come outside of the if-block. It's quite fast, and we can imagine a future use case where a user wants to try out different trimming parameters for the same model.